templates/client-daemonset.yaml

```diff
@@ -70,10 +70,18 @@ spec:
         {{- if .Values.global.tls.enabled }}
         - name: tls-ca-cert
           secret:
+            {{- if .Values.global.tls.caCert.secretName }}
+            secretName: {{ .Values.global.tls.caCert.secretName }}
+            {{- else }}
             secretName: {{ template "consul.fullname" . }}-ca-cert
+            {{- end }}
```

Comment:
If we do
```yaml
  volumes:
    - name: tls-ca-cert
      secret:
        {{- if .Values.global.tls.caCert.secretName }}
        secretName: {{ .Values.global.tls.caCert.secretName }}
        {{- else }}
        secretName: {{ template "consul.fullname" . }}-ca-cert
        {{- end }}
        items:
        - key: {{ default "tls.crt" .Values.global.tls.caCert.secretKey }}
          path: tls.crt
```

Then the ca cert will always end up in the file `tls.crt`. What do you think about that pattern instead of defaulting the `secretKey` to `tls.crt` in `values.yaml`? I think I like it better because the file system always looks the same, regardless of how it's being initialized.

Reply:
I really like this idea, thanks! I've updated everything to use this pattern instead.